# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
+ Formulas for the Bernoulli distribution: https://en.wikipedia.org/wiki/Bernoulli_distribution

In [18]:
import pandas as pd
import numpy as np
from scipy import stats


In [3]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [4]:
# number of callbacks for black-sounding names
wcall=sum(data[data.race=='w'].call)
bcall=sum(data[data.race=='b'].call)
print(wcall)



235.0


In [21]:
w = data[data.race=='w']
b = data[data.race=='b']


wn=len(w)
bn=len(b)

wp=wcall/wn
bp=bcall/bn

print(wp)
print(bp)
print(bn*bp)
#Greater than 5 and all 1-p combination or other p would be lower so all n*p are greater than 5. 




0.09650924024640657
0.06447638603696099
157.0


<div class="span5 alert alert-success">
<p>because n>30 and 235 and 157 respectivly are less than 10% of total population CLT works here and we will thus use a 2 sample z sided test since t is used when n<30. H0 is that white rate-black rate=0 while HA is the white rate-black rate!=0</p>
</div>


In [6]:
w = data[data.race=='w']
b = data[data.race=='b']

In [33]:
bootsize=9000
bootrep=np.empty(2435*2)
bootrep[0:2434]=1
bootrep[2435:4879]=0
bootdiff=np.empty(bootsize)
for x in range(bootsize):
   wrep=np.sum(np.random.choice(bootrep, size=wn))
   brep=np.sum(np.random.choice(bootrep, size=bn))
   bootdiff[x]=(wrep-brep)/2435
bootp=np.sum(bootdiff>(wp/bp))/bootsize
print(bootp) #Bootstrap P-Value
bootci=np.percentile(bootdiff, [2.5,97.5])
print(bootci) #Bootstrap Confidence Interval
bootme=bootci[1]-np.mean(bootdiff)
print(bootme) # Bootstrap Margin of Error



0.0
[-0.02874743  0.02833676]
0.02830203057266712


In [39]:
wp #phat white (1)
bp #phat black (2)
p=(wcall+bcall)/(wn+bn) #phat denominator 
print(p)

ztest=(wp-bp-0)/np.sqrt(((p*(1-p))/wn)+((p*(1-p))/bn))

freqp = stats.norm.sf(abs(ztest))*2
print(ztest, freqp ) #Freq Ztest and P-Value

chigh=(wp-bp)+1.96*(np.sqrt(((wp*(1-wp))/wn)+((bp*(1-bp))/bn)))
clow=(wp-bp)-1.96*(np.sqrt(((wp*(1-wp))/wn)+((bp*(1-bp))/bn)))
print(clow, chigh) #Freq Confidence Interval

freqmoe=(chigh-clow)/2.0
print(freqmoe) #Freq Margin of Error

0.08049281314168377
4.108412152434346 3.983886837585077e-05
0.016777447859559147 0.047288260559332024
0.015255406349886438


<div class="span5 alert alert-success">
<p> It is statistically significant under both models that white sounding names are more likely to receive a call back than black sounding names according to both models with p values significantly under the threshold of .05. No because its the only variable that we looked at while ignoring the other 30+ variables in the data set which could have cause more impacts.  </p>
</div>